go-controller/pkg/allocator/pod/pod_annotation.go

```diff
@@ -389,7 +397,7 @@ func allocatePodAnnotationWithRollback(
 		podAnnotation = tentative
 	}
 
-	if hasIPAMClaim && needsAnnotationUpdate && err == nil {
+	if ipamClaim != nil && claimsReconciler != nil && err == nil {
```

Comment:
nit: `ipamClaim != nil` can't be if `claimsReconciler == nil`

Reply:
Done.